hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/acl/OzoneObjInfo.java

```diff
@@ -69,6 +72,32 @@ public String getKeyName() {
     return keyName;
   }
 
+  public static OzoneObjInfo fromProtobuf(OzoneManagerProtocolProtos.OzoneObj
+      proto) {
+    Builder builder = new Builder()
+        .setResType(ResourceType.valueOf(proto.getResType().name()))
+        .setStoreType(StoreType.valueOf(proto.getStoreType().name()));
+    StringTokenizer tokenizer = new StringTokenizer(proto.getPath(),
+        OzoneConsts.OZONE_URI_DELIMITER);
+    // Set volume name.
+    if (tokenizer.hasMoreTokens()) {
+      builder.setVolumeName(tokenizer.nextToken());
+    }
+    // Set bucket name.
+    if (tokenizer.hasMoreTokens()) {
+      builder.setBucketName(tokenizer.nextToken());
+    }
+    // Set key name
+    StringBuffer sb = new StringBuffer();
```

Comment:
should we wrap line 91-97 with if (tokenizer.hasMoreTokens()) to avoid setting empty keyName when keyName is not needed.

Reply:
done